tests/SprykerEcoTest/Zed/Adyen/Business/Facade/ExecutePostSaveHookTest.php

```diff
@@ -0,0 +1,374 @@
+
+        $orderTransfer = $this->setUpCommandTest(
+            static::PROCESS_NAME_ADYEN_CREDIT_CARD,
+            static::OMS_STATUS_NEW,
+        );
+
+        $checkoutResponseTransfer = new CheckoutResponseTransfer();
+
+        // Act
+        $adyenFacade->executePostSaveHook(
+            $this->tester->buildQuoteTransfer($this->getPaymentAdyenReference($orderTransfer->getIdSalesOrder())),
+            $checkoutResponseTransfer,
+        );
+
+        // Assert
+        $this->assertCount(1, $checkoutResponseTransfer->getErrors());
+
+        $this->assertFalse($checkoutResponseTransfer->getIsSuccess());
+
+        $checkoutErrorTransfer = $checkoutResponseTransfer->getErrors()->getIterator()->current();
+        /** @var \Generated\Shared\Transfer\CheckoutErrorTransfer $checkoutErrorTransfer */
+        $this->assertSame(static::ERROR_CODE_PAYMENT_FAILED, $checkoutErrorTransfer->getErrorCode());
+        $this->assertSame(static::ERROR_TYPE_PAYMENT_FAILED, $checkoutErrorTransfer->getErrorType());
+
+        $paymentAdyenApiLogEntity = $this->findLastPaymentAdyenApiLog();
+        $this->assertNotNull($paymentAdyenApiLogEntity);
+        $this->assertSame(static::MAKE_PAYMENT_PAY_PAL_REQUEST_TYPE, $paymentAdyenApiLogEntity->getType());
+        $this->assertTrue($paymentAdyenApiLogEntity->getIsSuccess());
+
+        $paymentAdyenEntity = $this->findPaymentAdyen($orderTransfer);
+        $this->assertNotNull($paymentAdyenEntity);
+        $this->assertSame(static::MAKE_PAYMENT_RESPONSE_PAYMENT_DATA, $paymentAdyenEntity->getPaymentData());
+        $this->assertSame(json_encode(static::MAKE_PAYMENT_RESPONSE_DETAILS), $paymentAdyenEntity->getDetails());
+    }
+
+    /**
+     * @return void
+     */
+    public function testExecutePostSaveHookWithoutMakePaymentResponse(): void
+    {
+        // Arrange
+        $adyenApiResponseTransfer = (new AdyenApiResponseBuilder([
+                AdyenApiResponseTransfer::IS_SUCCESS => true,
+            ]))
+            ->build();
+
+        $adyenToAdyenApiFacadeBridgeMock = $this->createMock(AdyenToAdyenApiFacadeBridge::class);
+        $adyenToAdyenApiFacadeBridgeMock->method('performMakePaymentApiCall')
+            ->willReturn($adyenApiResponseTransfer);
+
+        $adyenFacade = $this->createFacade($adyenToAdyenApiFacadeBridgeMock);
+
+        $orderTransfer = $this->setUpCommandTest(
+            static::PROCESS_NAME_ADYEN_CREDIT_CARD,
+            static::OMS_STATUS_NEW,
+        );
+
+        $checkoutResponseTransfer = new CheckoutResponseTransfer();
+
+        // Assert
+        $this->expectException(NullValueException::class);
+
+        // Act
+        $adyenFacade->executePostSaveHook(
+            $this->tester->buildQuoteTransfer($this->getPaymentAdyenReference($orderTransfer->getIdSalesOrder())),
+            $checkoutResponseTransfer,
+        );
+    }
+
+    /**
+     * @return void
+     */
+    public function testExecutePostSaveHookWithoutPayment(): void
```

Comment:
pattern for method naming is `test{Method}{Result}{Conditions}`
so `testExecutePostSaveHookThrowsExceptionWhenPaymentIsMissing`
`ExecutePostSaveHook` part could be skipped since it's already described at class naming (you have test class per method and not for the whole facade)
so `testThrowsExceptionWhenPaymentIsMissing` or `testThrowsExceptionWhenPaymentIsNotProvided` is enough

Reply:
fixed naming for all methods in this class